src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -178,7 +178,13 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
   }
 
   private buildResultPreview(suggestionText: string, element: HTMLElement, rank: number): ISearchResultPreview {
-    $$(element).addClass('coveo-preview-selectable');
+    element.classList.add('coveo-preview-selectable');
+    const resultLink = element.querySelector(Component.computeSelectorForType(ResultLink.ID)) as HTMLElement;
+    if (resultLink) {
+      element.setAttribute('aria-label', resultLink.textContent);
+      resultLink.setAttribute('role', 'link');
+      resultLink.removeAttribute('aria-level');
```

Comment:
Why is this needed ?

Reply:
It seems result link by default has a aria-level of 2 and a role of header if it's the first result link.
@btaillon were those added to one of the result links?

Reply:
Ah, I didn't notice that this only affects the first result link. I'll check it out. But yes, I did add this because of the header role and aria-level of 2, which caused Axe to give some warnings.